### ISSUE TITLE
Alternative wait on a set of dynamic set of Ballerina future values

### ISSUE THREAD

**Issue by NipunaRanasinghe** [ballerina-platform/ballerina-spec#888]
**Description:**
Do we have any plans on adding language-level support to alternatively wait on a given dynamic set of futures (i.e. can be provided as an array/rest args) and return the result? 

@rdhananjaya has suggested me the below function which can emulate the expected functionality. Just wanted to know whether we have any plans on adding this support at the language-level in near future.

```ballerina
function waitHelper(future<any>[] fs) returns any|error {
    match fs {
        var [f] => {
            return wait f;
        }
        var [f, x] => {
            return wait f | x;
        }
        var [f, x, y] => {
            return wait f | x | y;
        }
        var [f, x, y, ...r] => {
            var rest = start waitHelper(r);
            return wait f | x | y | rest;
        }
    }
}
```

**Suggested Labels:**

   
**Code sample that shows issue:**

**Related Issues:**

**Comment by jclark** [comment]
No immediate plans, but if there is a user requirement then we can prioritise it.

Is this related to #104?

**Comment by NipunaRanasinghe** [comment]
> Is this related to #104?

@jclark Yes it seems so to me. Also I found this issue (https://github.com/ballerina-platform/ballerina-lang/issues/25969) in ballerina-lang repo, which was raised by @ramith sometimes back.
